### PR TITLE
Support for execute all test command

### PR DESCRIPTION
**Pull request by rafaelbey**
AS part of supporting a command to execute all test cases, this PR also introduce the following changes

- Execution results, like those from test cases can carry an optional location.  This is to be used on the front end to provide context on the result and allow navigation to it
- Support for function test cases, as functions are now testable
- Refactor tds request execution for better segregation of LSP vs extension code
- Introduce a custom JRPC for legend specific APIs, starting with the tds function requests
- Refactor integration test, so we can reuse the setup rather than ending with one monolithic test
- Add version properties, to help understanding the versions used when the server was compiled

### CODE REVIEW

File: legend-engine-ide-lsp-server/src/main/java/org/finos/legend/engine/ide/lsp/server/LegendLanguageServer.java [hunk 1]
```diff
@@ -1057,4 +1068,23 @@ public static void main(String[] args)
...
+        {
+            if (stream != null)
+            {
+                properties.load(stream);
+                return properties;
+            }
+        }
+        catch (Exception e)
+        {
+            e.printStackTrace();
```
Comment (kevin-m-knight-gs) [inline legend-engine-ide-lsp-server/src/main/java/org/finos/legend/engine/ide/lsp/server/LegendLanguageServer.java:1086]: We should either log this or throw

File: legend-engine-ide-lsp-server/src/main/java/org/finos/legend/engine/ide/lsp/server/LegendLanguageServerContract.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
+// Copyright 2023 Goldman Sachs
```
Comment (kevin-m-knight-gs) [inline legend-engine-ide-lsp-server/src/main/java/org/finos/legend/engine/ide/lsp/server/LegendLanguageServerContract.java:1]:
```suggestion
// Copyright 2024 Goldman Sachs
```

**Review by kevin-m-knight-gs — CHANGES_REQUESTED** [review]

File: legend-engine-ide-lsp-server/src/main/java/org/finos/legend/engine/ide/lsp/server/LegendLanguageServerContract.java (hunk 2 above)
Comment (rafaelbey) [reply legend-engine-ide-lsp-server/src/main/java/org/finos/legend/engine/ide/lsp/server/LegendLanguageServerContract.java]: Done!

File: legend-engine-ide-lsp-server/src/main/java/org/finos/legend/engine/ide/lsp/server/LegendLanguageServer.java (hunk 1 above)
Comment (rafaelbey) [reply legend-engine-ide-lsp-server/src/main/java/org/finos/legend/engine/ide/lsp/server/LegendLanguageServer.java]: Added to log.  I dont want the process to fail as this is for information purposes so it easy to track potential exceptions from incompatible versions (we built with engine 4.40, but user is trying 4.50 and that could not be backward compatible)

**Pull request merged**